packages/block-library/src/image/edit.js

```diff
@@ -293,8 +303,13 @@ class ImageEdit extends Component {
 		this.props.setAttributes( { ...extraUpdatedAttributes, align: nextAlign } );
 	}
 
-	updateImageURL( url ) {
-		this.props.setAttributes( { url, width: undefined, height: undefined } );
+	updateImageURL( url, data ) {
+		this.props.setAttributes( {
+			url,
+			width: undefined,
+			height: undefined,
+			mediaSizeClass: 'size-' + data.slug,
```

Comment:
Can we store `mediaSize` and prefix it with `size-` when applying classes to HTML elements?